universal-application-tool-0.0.1/app/auth/oidc/IdcsOidcProvider.java

```diff
@@ -0,0 +1,59 @@
+package auth.oidc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.ProfileFactory;
+import com.google.inject.Inject;
+import javax.inject.Provider;
+import org.pac4j.core.http.callback.PathParameterCallbackUrlResolver;
+import org.pac4j.oidc.client.OidcClient;
+import org.pac4j.oidc.config.OidcConfiguration;
+import repository.UserRepository;
+
+public class IdcsOidcProvider implements Provider<OidcClient> {
+
+  private final com.typesafe.config.Config configuration;
+  private final String baseUrl;
+  private final ProfileFactory profileFactory;
+  private final Provider<UserRepository> applicantRepositoryProvider;
+
+  @Inject
+  public IdcsOidcProvider(
+      com.typesafe.config.Config configuration,
+      ProfileFactory profileFactory,
+      Provider<UserRepository> applicantRepositoryProvider) {
+    this.configuration = checkNotNull(configuration);
+    this.baseUrl = configuration.getString("base_url");
```

Comment:
question (blocking): are we validating that `base_url` is set anywhere?

Reply:
I realized we actually don't, anywhere in the code base. How do you think I should handle it-- should we throw an exception in these provider classes if it's not set? Would it make more sense to throw an exception from somewhere else? I actually think we have a default setting for it in application.conf so maybe we don't need to check it? Curious your thoughts

Reply:
It's defaulted to `localhost:9000` right? my worry here is that someone deploying to a staging or production instance might forget to set this value, and then the authentication won't work because the callback is wrong. @bion @shanemc-goog what are your thoughts on this?